Bottom: 32bc4add78916f06ef1740dac6e09db624541999
Top:    1942727d83d8b3be2e14fc20a67432ae618a4d08
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-08-26 14:31:54 +0100

Refresh of hid-gtk

---

diff --git a/src/hid/gtk/gtkhid-main.c b/src/hid/gtk/gtkhid-main.c
index 2e258d1..f5deb16 100644
--- a/src/hid/gtk/gtkhid-main.c
+++ b/src/hid/gtk/gtkhid-main.c
@@ -72,15 +72,6 @@ ghid_pan_view_abs (Coord pcb_x, Coord pcb_y, int widget_x, int widget_y)
   pan_common (gport);
 }
 
-void
-ghid_pan_view_rel (Coord dx, Coord dy)
-{
-  gport->view_x0 += dx;
-  gport->view_y0 += dy;
-
-  pan_common (gport);
-}
-
 
 /* gport->zoom:
  * zoom value is PCB units per screen pixel.  Larger numbers mean zooming
@@ -1848,17 +1839,17 @@ ScrollAction (int argc, char **argv, Coord x, Coord y)
     div = atoi(argv[1]);
 
   if (strcasecmp (argv[0], "up") == 0)
-    dy = -gport->view_height / div;
+    dy = -(ghid_port.height * gport->zoom / div);
   else if (strcasecmp (argv[0], "down") == 0)
-    dy = gport->view_height / div;
+    dy = ghid_port.height * gport->zoom / div;
   else if (strcasecmp (argv[0], "right") == 0)
-    dx = gport->view_width / div;
+    dx = ghid_port.width * gport->zoom / div;
   else if (strcasecmp (argv[0], "left") == 0)
-    dx = -gport->view_width / div;
+    dx = -(ghid_port.width * gport->zoom / div);
   else
     AFAIL (scroll);
 
-  ghid_pan_view_rel (dx, dy);
+  ghid_port_ranges_pan (dx, dy, TRUE);
 
   return 0;
 }
